services/catarse/app/models/billing/payment.rb

```diff
@@ -48,6 +48,22 @@ def installment?
       installments_count > 1
     end
 
+    scope :all_boleto_that_should_be_refused, lambda {
+      Billing::Payment.joins(:boleto).where(
+        'billing_boletos.expires_on < current_timestamp',
+        payment_method: Billing::PaymentMethods::BOLETO,
+        state: 'waiting_payment'
+      )
+    }
+
+    scope :all_pix_that_should_be_refused, lambda {
+      Billing::Payment.joins(:pix).where(
+        'billing_pixes.expires_on < current_timestamp',
+        payment_method: Billing::PaymentMethods::PIX,
+        state: 'waiting_payment'
+      )
+    }
+
```

Comment:
**Sugestão**
Separar isso num query object, igual como fizemos no cartão de crédito, pois se formos usar as mesmas práticas presentes no código legado do projeto, essa classe aqui vai ficar igual como era antes da refatoração e nosso trabalho será em vão.

Tentar colocar essa query em apenas uma, que retorne os pagamentos vencidos, pois no contexto dessa atividade não temos necessidade de saber essa informação de forma separado, haja visto que o código de expiração está repetido para os dois casos.

Quando for filtrar os objetos pelo estado da State Machine, utilizar o `in_state` pois é mais seguro e correto, do que usar apenas a coluna.